org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/AbstractImageBuilder.java

```diff
@@ -219,8 +221,11 @@ public void acceptResult(CompilationResult result) {
 					createProblemFor(compilationUnit.resource, null, Messages.build_inconsistentClassFile, JavaCore.ERROR);
 			}
 		}
-		if (result.hasAnnotations && this.filesWithAnnotations != null) // only initialized if an annotation processor is attached
-			this.filesWithAnnotations.add(compilationUnit);
+		if (result.hasAnnotations && this.filesWithAnnotations != null) {
```

Comment:
hasAnnotations is set even when the result is an annotation, in which case, we add this to the map even when the AnnotationBinding[] is empty. I believe we can avoid this being added to the map?

Reply:
So your suggestion is tho also check for `annotationbinding.length!=0`?

Reply:
Yes